src/UniswapV4Migrator.sol

```diff
@@ -237,136 +237,213 @@ contract UniswapV4Migrator is ILiquidityMigrator, ImmutableAirlock {
+                    new bytes(0)
+                );
+            }
 
-            params[1] = abi.encode(poolKey.currency0, poolKey.currency1);
+            if (abovePriceLiquidity > 0) {
+                params[paramsIndex++] = abi.encode(
+                    poolKey,
+                    currentTick + poolKey.tickSpacing,
+                    upperTick,
+                    uint128(abovePriceLiquidity),
+                    uint128(balance0),
+                    0,
+                    address(this),
+                    new bytes(0)
+                );
+            }
         } else {
             // Standard case: split liquidity 10/90
-            uint256 lockedLiquidity = liquidity / 10;
-            uint256 timeLockLiquidity = liquidity - lockedLiquidity;
-
-            // Liquidity for the protocol locker
-            params[0] = abi.encode(
-                poolKey,
-                lowerTick,
-                upperTick,
-                uint128(lockedLiquidity),
-                uint128(balance0) / 10,
-                uint128(balance1) / 10,
-                address(this),
-                new bytes(0)
-            );
+            if (belowPriceLiquidity > 0) {
+                uint160 lockedBelowPriceLiquidity = LiquidityAmounts.getLiquidityForAmounts(
+                    sqrtPriceX96,
+                    TickMath.getSqrtPriceAtTick(lowerTick),
+                    TickMath.getSqrtPriceAtTick(currentTick - poolKey.tickSpacing),
+                    0,
+                    uint128(balance1) / 10
+                );
 
-            // Liquidity for the Timelock, we can pass the full balances as maximum
-            // amounts here since the protocol locker already received its share
-            params[1] = abi.encode(
-                poolKey,
-                lowerTick,
-                upperTick,
-                uint128(timeLockLiquidity),
-                uint128(balance0),
-                uint128(balance1),
-                recipient,
-                new bytes(0)
-            );
+                uint256 timeLockBelowPriceLiquidity = belowPriceLiquidity - lockedBelowPriceLiquidity;
+
+                // Liquidity for the protocol locker
+                params[paramsIndex++] = abi.encode(
+                    poolKey,
+                    lowerTick,
+                    currentTick - poolKey.tickSpacing,
+                    uint128(lockedBelowPriceLiquidity),
+                    0,
+                    uint128(balance1) / 10,
+                    address(this),
+                    new bytes(0)
+                );
+
+                // Liquidity for the Timelock, we can pass the full balances as maximum
+                // amounts here since the protocol locker already received its share
+                params[paramsIndex++] = abi.encode(
+                    poolKey,
+                    lowerTick,
+                    currentTick - poolKey.tickSpacing,
+                    uint128(timeLockBelowPriceLiquidity),
```

Comment:
Should we also decrement by one here for the same reason we do on line 397? I would expect the same risk to be present

Actually I think it would be even safer to decrement the balance provided to `getLiquidityForAmounts` in each of `belowPriceLiquidity`, `abovePriceLiquidity`,`lockedBelowPriceLiquidity`, and `lockedAbovePriceLiquidity` to be extra cautious